File: algorithms/0383.ransom-note/solution.rb
# @param {String} ransom_note
# @param {String} magazine
# @return {Boolean}
def can_construct(ransom_note, magazine)
  mag_chars = magazine.chars
  can = true
  ransom_note.chars do |c|
    idx = mag_chars.find_index(c)
    if idx
      mag_chars.delete_at(idx)
    else
      break can = false
    end
  end
  can
end

File: algorithms/0383.ransom-note/solution.rs
impl Solution {
    pub fn can_construct(ransom_note: String, magazine: String) -> bool {
        let mut mag_chars: Vec<char> = magazine.chars().collect();
        for c in ransom_note.chars() {
            if let Some(idx) = mag_chars.iter().position(|x| x == &c) {
                mag_chars.remove(idx);
            } else {
                return false;
            }
        }

        true
    }
}
